src/mk.rs

```diff
@@ -402,17 +401,17 @@ impl Builder {
         rng: &mut (impl RngCore + CryptoRng),
     ) -> Result<Self, Error> {
         let key_bytes = match codec {
-            Codec::Ed25519Priv => Ed25519Keypair::random(rng).private.to_bytes().to_vec(),
+            Codec::Ed25519Priv => ed25519_dalek::SigningKey::generate(rng).to_bytes().to_vec(),
```

Comment:
When I made this change, I was in the process of placing `ssh` behind a feature flag, the change was needed otherwise users didn't have access to Ed25519 since `Ed25519Keypair` comes from `ssh_key::private::Ed25519Keypair`. `ssh-key` sources from `dalek` anyway, so it's the same dep. 

Once I shifted to the `ssh-key defaults = false` instead of an `ssh` feature flag, that reason is no longer valid. 

That said, if this crate ever uses feature flags for `ssh` (maybe it still should? not everyone may want the extra dependency?) then the replacement would be valid again.

I don't see an impact either way at this point, pending further discussion on feature flags.